cores/arduino/Uart.cpp

```diff
@@ -102,11 +129,30 @@ int Uart::peek()
 
 int Uart::read()
 {
-  return rxBuffer.read_char();
+  int c = rxBuffer.read_char();
+
+  if (uc_pinRTS != NO_RTS_PIN) {
+    // if there is enough space in the RX buffer, assert RTS
+    if (rxBuffer.availableForStore() > RTS_RX_THRESHOLD) {
+      digitalWrite(uc_pinRTS, LOW);
+    }
+  }
+
+  return c;
 }
 
 size_t Uart::write(const uint8_t data)
 {
+  if (uc_pinRTS != NO_RTS_PIN) {
```

Comment:
why? should RTS influence only Rx?

Reply:
Right, I misunderstood the RTS term - corrected!